graph/multi/multi.go

```diff
@@ -54,41 +48,42 @@ func (l Line) ID() int64 { return l.UID }
 
 // WeightedEdge is a collection of weighted multigraph edges sharing end points.
 type WeightedEdge struct {
-	Lines []graph.WeightedLine
+	F, T graph.Node
+
+	graph.WeightedLines
 
 	// WeightFunc calculates the aggregate
 	// weight of the lines in Lines. If
 	// WeightFunc is nil, the sum of weights
 	// is used as the edge weight.
-	WeightFunc func([]graph.WeightedLine) float64
+	// The graph.WeightedLines cam be expected
+	// to be positioned at the first line og
+	// the iterator on entry and must be
+	// Reset before exit.
+	// WeightFunc must accept a nil input.
+	WeightFunc func(graph.WeightedLines) float64
 }
 
 // From returns the from-node of the edge.
-func (e WeightedEdge) From() graph.Node {
-	if len(e.Lines) == 0 {
-		return nil
-	}
-	return e.Lines[0].From()
-}
+func (e WeightedEdge) From() graph.Node { return e.F }
 
 // To returns the to-node of the edge.
-func (e WeightedEdge) To() graph.Node {
-	if len(e.Lines) == 0 {
-		return nil
-	}
-	return e.Lines[0].To()
-}
+func (e WeightedEdge) To() graph.Node { return e.T }
 
 // Weight returns the weight of the edge.
 func (e WeightedEdge) Weight() float64 {
+	if e.WeightedLines == nil {
+		return 0
+	}
 	if e.WeightFunc == nil {
 		var w float64
-		for _, l := range e.Lines {
-			w += l.Weight()
+		for e.Next() {
```

Comment:
Is it ok to assume that `e` is at the first line? Above it is documented for the input to `WeightFunc` but what about `Weight`?

Reply:
Added documentation making this clear.